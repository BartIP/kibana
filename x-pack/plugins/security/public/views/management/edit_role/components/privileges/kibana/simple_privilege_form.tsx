/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import {
  // @ts-ignore
  EuiDescribedFormGroup,
  EuiFormRow,
} from '@elastic/eui';
<<<<<<< HEAD
=======
import { FormattedMessage } from '@kbn/i18n/react';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import React, { Component, Fragment } from 'react';
import { KibanaPrivilege } from '../../../../../../../common/model/kibana_privilege';
import { Role } from '../../../../../../../common/model/role';
import { isReservedRole } from '../../../../../../lib/role';
import { NO_PRIVILEGE_VALUE } from '../../../lib/constants';
import { copyRole } from '../../../lib/copy_role';
import { PrivilegeSelector } from './privilege_selector';

interface Props {
  kibanaAppPrivileges: KibanaPrivilege[];
  role: Role;
  onChange: (role: Role) => void;
  editable: boolean;
}

export class SimplePrivilegeForm extends Component<Props, {}> {
  public render() {
    const { kibanaAppPrivileges, role } = this.props;

    const assignedPrivileges = role.kibana;

    const kibanaPrivilege: KibanaPrivilege =
      assignedPrivileges.global.length > 0
        ? (assignedPrivileges.global[0] as KibanaPrivilege)
        : NO_PRIVILEGE_VALUE;

<<<<<<< HEAD
    const description = <p>Specifies the Kibana privilege for this role.</p>;

    return (
      <Fragment>
        <EuiDescribedFormGroup title={<h3>Kibana privileges</h3>} description={description}>
=======
    const description = (
      <p>
        <FormattedMessage
          id="xpack.security.management.editRoles.simplePrivilegeForm.specifyPrivilegeForRoleDescription"
          defaultMessage="Specifies the Kibana privilege for this role."
        />
      </p>
    );

    return (
      <Fragment>
        <EuiDescribedFormGroup
          title={
            <h3>
              <FormattedMessage
                id="xpack.security.management.editRoles.simplePrivilegeForm.kibanaPrivilegesTitle"
                defaultMessage="Kibana privileges"
              />
            </h3>
          }
          description={description}
        >
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
          <EuiFormRow hasEmptyLabelSpace>
            <PrivilegeSelector
              data-test-subj={'kibanaPrivilege'}
              availablePrivileges={kibanaAppPrivileges}
              value={kibanaPrivilege}
              disabled={isReservedRole(role)}
              allowNone={true}
              onChange={this.onKibanaPrivilegeChange}
            />
          </EuiFormRow>
        </EuiDescribedFormGroup>
      </Fragment>
    );
  }

  public onKibanaPrivilegeChange = (privilege: KibanaPrivilege) => {
    const role = copyRole(this.props.role);

    // Remove base privilege value
    role.kibana.global = [];

    if (privilege !== NO_PRIVILEGE_VALUE) {
      role.kibana.global = [privilege];
    }

    this.props.onChange(role);
  };
}
