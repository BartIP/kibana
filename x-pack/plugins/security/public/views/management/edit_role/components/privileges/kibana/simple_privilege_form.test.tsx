/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

<<<<<<< HEAD
import { mount, shallow } from 'enzyme';
import React from 'react';
=======
import React from 'react';
import { mountWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { PrivilegeSelector } from './privilege_selector';
import { SimplePrivilegeForm } from './simple_privilege_form';

const buildProps = (customProps?: any) => {
  return {
    role: {
      name: '',
      elasticsearch: {
        cluster: ['manage'],
        indices: [],
        run_as: [],
      },
      kibana: {
        global: [],
        space: {},
      },
    },
    editable: true,
    kibanaAppPrivileges: ['all', 'read'],
    onChange: jest.fn(),
    ...customProps,
  };
};

describe('<SimplePrivilegeForm>', () => {
  it('renders without crashing', () => {
<<<<<<< HEAD
    expect(shallow(<SimplePrivilegeForm {...buildProps()} />)).toMatchSnapshot();
=======
    expect(shallowWithIntl(<SimplePrivilegeForm {...buildProps()} />)).toMatchSnapshot();
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  });

  it('displays "none" when no privilege is selected', () => {
    const props = buildProps();
<<<<<<< HEAD
    const wrapper = shallow(<SimplePrivilegeForm {...props} />);
=======
    const wrapper = shallowWithIntl(<SimplePrivilegeForm {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    const selector = wrapper.find(PrivilegeSelector);
    expect(selector.props()).toMatchObject({
      value: 'none',
    });
  });

  it('displays the selected privilege', () => {
    const props = buildProps({
      role: {
        elasticsearch: {},
        kibana: {
          global: ['read'],
        },
      },
    });
<<<<<<< HEAD
    const wrapper = shallow(<SimplePrivilegeForm {...props} />);
=======
    const wrapper = shallowWithIntl(<SimplePrivilegeForm {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    const selector = wrapper.find(PrivilegeSelector);
    expect(selector.props()).toMatchObject({
      value: 'read',
    });
  });

  it('fires its onChange callback when the privilege changes', () => {
    const props = buildProps();
<<<<<<< HEAD
    const wrapper = mount(<SimplePrivilegeForm {...props} />);
=======
    const wrapper = mountWithIntl(<SimplePrivilegeForm {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    const selector = wrapper.find(PrivilegeSelector).find('select');
    selector.simulate('change', { target: { value: 'all' } });

    expect(props.onChange).toHaveBeenCalledWith({
      name: '',
      elasticsearch: {
        cluster: ['manage'],
        indices: [],
        run_as: [],
      },
      kibana: {
        global: ['all'],
        space: {},
      },
    });
  });
});
