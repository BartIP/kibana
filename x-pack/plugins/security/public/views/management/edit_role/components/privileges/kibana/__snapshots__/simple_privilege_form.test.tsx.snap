// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimplePrivilegeForm> renders without crashing 1`] = `
<React.Fragment>
  <EuiDescribedFormGroup
    description={
      <p>
<<<<<<< HEAD
        Specifies the Kibana privilege for this role.
=======
        <FormattedMessage
          defaultMessage="Specifies the Kibana privilege for this role."
          id="xpack.security.management.editRoles.simplePrivilegeForm.specifyPrivilegeForRoleDescription"
          values={Object {}}
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </p>
    }
    fullWidth={false}
    gutterSize="l"
    title={
      <h3>
<<<<<<< HEAD
        Kibana privileges
=======
        <FormattedMessage
          defaultMessage="Kibana privileges"
          id="xpack.security.management.editRoles.simplePrivilegeForm.kibanaPrivilegesTitle"
          values={Object {}}
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </h3>
    }
    titleSize="xs"
  >
    <EuiFormRow
      describedByIds={Array []}
      fullWidth={false}
      hasEmptyLabelSpace={true}
    >
      <PrivilegeSelector
        allowNone={true}
        availablePrivileges={
          Array [
            "all",
            "read",
          ]
        }
        data-test-subj="kibanaPrivilege"
        disabled={false}
        onChange={[Function]}
        value="none"
      />
    </EuiFormRow>
  </EuiDescribedFormGroup>
</React.Fragment>
`;
